src/node/stage/approved.rs

```diff
@@ -676,10 +678,19 @@ impl Approved {
             .prefix
             .is_neighbour(self.shared_state.our_prefix())
         {
-            self.cast_unordered_vote(core, Vote::SectionInfo(elders_info))?;
+            self.cast_unordered_vote(core, Vote::SectionInfo(elders_info.clone()))?;
         }
 
-        Ok(())
+        if let Some(dst_key) = dst_key {
+            if self.shared_state.our_history.has_key(&dst_key)
+                && dst_key == *self.shared_state.our_history.last_key()
```

Comment:
`dst_key == our_history.last_key()` implies `our_history.has_key(dst_key)`, so the first check is redundant.